src/System.Net.Http/src/System/Net/Http/Headers/HeaderDescriptor.cs

```diff
@@ -99,19 +102,58 @@ public string GetHeaderValue(ReadOnlySpan<byte> headerValue)
             }
 
             // If it's a known header value, use the known value instead of allocating a new string.
-            if (_knownHeader != null && _knownHeader.KnownValues != null)
+            if (_knownHeader != null)
             {
-                string[] knownValues = _knownHeader.KnownValues;
-                for (int i = 0; i < knownValues.Length; i++)
+                if (_knownHeader.KnownValues != null)
+                {
+                    string[] knownValues = _knownHeader.KnownValues;
+                    for (int i = 0; i < knownValues.Length; i++)
+                    {
+                        if (ByteArrayHelpers.EqualsOrdinalAsciiIgnoreCase(knownValues[i], headerValue))
+                        {
+                            return knownValues[i];
+                        }
+                    }
+                }
+
+                if (KnownHeader == KnownHeaders.Location)
```

Comment:
"KnownHeader" is a type.  So, this doesn't look right.  I think this should be:

```c#
if (_knownHeader == KnownHeaders.Location)
```

Reply:
This is both property and type - compiler seems to disambiguate this correctly (also tests would have failed)

Reply:
Consistency -- the rest of the code in this file uses `_knownHeader`.